src/field/arena.ts

```diff
@@ -301,8 +301,8 @@ export class Arena {
 
   /**
    * Attempts to set a new weather to the battle
-   * @param weather new weather to set of type WeatherType
-   * @param hasPokemonSource is the new weather from a pokemon
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
+   * @param hasPokemonSource boolean if the new weather is from a pokemon
```

Comment:
```suggestion
   * @param hasPokemonSource true if the new weather is from a pokemon
```

you don’t need to put types into JS doc as we are using ts. This is just redundant

Reply:
I find them quite useful though for referring to types in vscode, for example in this case vscode gives you a direct link to the `WeatherType` enum, without the linkcode you have to take an extra step to search/find it

![screenshot](https://github.com/user-attachments/assets/d89d27ec-20a0-4b01-84db-d2d04bae1c1c)

Reply:
No offense but there is no extra steps needed. Look above 
`(method) Arena.trySetWeather(weather:  WeatherType, hasPokemonSource: boolean)`

I understand that you mean the internal link but I didn't remove that for the WeatherType. I'm aware of that feature. I just rephrased it to be less redundant

Reply:
thanks, for clarifying i get you (remove extra text, no the linkcode itself)